### ISSUE TITLE
fix unit tests for clojure 1.7

### ISSUE THREAD

**Issue by loomis** [slipstream/SlipStreamUI#369]
With clojure 1.7.0-RC2 many of the SlipStreamUI unit tests fail.  These tests should be refactored so that they work on both clojure 1.6.0 and 1.7.0-RC2.  (Moved from sixsq/Tasklist#333.)

**Comment by loomis** [comment]
Tracebacks on the error messages look like:
```
expectations.junit.ExpectationsFailure$NullException: failure in (interop_test.clj:73) : slipstream.ui.util.interop-test
(expect
 (conj
  i-map-ancestors
  clojure.lang.AFn
  clojure.lang.APersistentMap
  clojure.lang.IEditableCollection
  clojure.lang.IFn
  clojure.lang.IHashEq
  clojure.lang.IMeta
  clojure.lang.IObj
  clojure.lang.IPersistentMap
  clojure.lang.MapEquivalence
  java.io.Serializable
  java.lang.Object
  java.lang.Runnable
  java.util.concurrent.Callable
  java.util.Map)
 (ancestors clojure.lang.PersistentArrayMap))

           expected: #{clojure.lang.IMeta clojure.lang.Counted java.util.concurrent.Callable java.lang.Runnable clojure.lang.IObj java.lang.Iterable clojure.lang.IEditableCollection clojure.lang.IFn clojure.lang.AFn java.io.Serializable clojure.lang.IPersistentCollection clojure.lang.IHashEq java.lang.Object clojure.lang.APersistentMap clojure.lang.IPersistentMap clojure.lang.MapEquivalence java.util.Map clojure.lang.ILookup clojure.lang.Associative clojure.lang.Seqable} 
                was: #{clojure.lang.IMeta clojure.lang.Counted java.util.concurrent.Callable java.lang.Runnable clojure.lang.IObj java.lang.Iterable clojure.lang.IEditableCollection clojure.lang.IFn clojure.lang.AFn java.io.Serializable clojure.lang.IPersistentCollection clojure.lang.IHashEq java.lang.Object clojure.lang.APersistentMap clojure.lang.IPersistentMap clojure.lang.MapEquivalence java.util.Map clojure.lang.IMapIterable clojure.lang.ILookup clojure.lang.Associative clojure.lang.Seqable}

           in expected, not actual: null
           in actual, not expected: #{clojure.lang.IMapIterable}

	at expectations.junit.ExpectationsFailure.<init>(ExpectationsFailure.java:11)
	at expectations.junit.runner$failure.invoke(runner.clj:22)
	at clojure.core$partial$fn__4529.invoke(core.clj:2502)
	at expectations$eval1451$fn__1452.invoke(expectations.clj:169)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at slipstream.ui.util.interop_test$expect996973422.invoke(interop_test.clj:73)
	at expectations$test_var$fn__1491.invoke(expectations.clj:229)
	at expectations$test_var.invoke(expectations.clj:225)
	at expectations$test_vars$fn__1526.invoke(expectations.clj:263)
	at expectations$create_context.invoke(expectations.clj:243)
	at expectations$test_vars.invoke(expectations.clj:263)
	at expectations$run_tests_in_vars.invoke(expectations.clj:275)
	at expectations.junit.runner$create_runner$fn__11715$fn__11718.invoke(runner.clj:65)
	at expectations.junit.runner$create_runner$fn__11715.invoke(runner.clj:60)
	at expectations.junit.runner.proxy$org.junit.runner.Runner$ff19274a.run(Unknown Source)
	at expectations.junit.ExpectationsTestRunner.run(ExpectationsTestRunner.java:24)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
```

**Comment by loomis** [comment]
I started looking at this while waiting for release jobs to run.

The first set of changes are just ordering fixes and work on both clojure 1.6 and 1.7.

The second ones (`interop_test.clj`) are 'fixes' for new interfaces added with 1.7.  I don't actually understand what these tests accomplish, but they pass on 1.7 with the changes.  Probably these should be removed, made conditional on the clojure version, or modified to ignore ancestors in `clojure.lang`.

**Comment by rbf** [comment]
For reference, we've found with @st the root cause of these failing tests: an assumption of the order of something that was not supposed to be ordered. And the order of this thing (a map returned by `select-keys`) changed on 1.7:

Before 1.7:
``` clojure
(seq (select-keys {:a 1 :b 2, :c 3} [ :a :b]))
; => ([:b 2] [:a 1])
```

After 1.7:
``` clojure
(seq (select-keys {:a 1 :b 2, :c 3} [ :a :b]))
; => ([:a 1] [:b 2])
```

[Now](https://github.com/slipstream/SlipStreamUI/blob/3cb2395f567d300403e94a552cb9e29130e307f9/clj/src/slipstream/ui/views/table.clj#L41) we explicit the sorting of the `seq`'ed map.
